src/lib.rs

```diff
@@ -214,6 +227,34 @@ impl<T: Columnation> ColumnStack<T> {
     }
 }
 
+impl<A: Columnation, B: Columnation> ColumnStack<(A, B)> {
+    /// Copies a destructured tuple `(A, B)` into this column stack.
+    ///
+    /// This serves situations where a tuple should be constructed from its constituents but not
+    /// not all elements are available as owned data.
+    ///
+    /// The element can be read by indexing
+    pub fn copy_destructured(&mut self, t1: &A, t2: &B) {
+        unsafe {
+            self.local.push(self.inner.copy_destructured(t1, t2));
+        }
+    }
+}
+
+impl<A: Columnation, B: Columnation, C: Columnation> ColumnStack<(A, B, C)> {
+    /// Copies a destructured tuple `(A, B, C)` into this column stack.
+    ///
+    /// This serves situations where a tuple should be constructed from its constituents but not
+    /// not all elements are available as owned data.
+    ///
+    /// The element can be read by indexing
+    pub fn copy_destructured(&mut self, r0: &A, r1: &B, r2: &C) {
+        unsafe {
+            self.local.push(self.inner.copy_destructured(r0, r1, r2));
+        }
+    }
+}
+
```

Comment:
Nit, but was this an example of a thing that could be generalized with `paste` as well?

Reply:
Converted to macro.